cpp/src/io/parquet/reader_impl.hpp

```diff
@@ -130,6 +130,15 @@ class reader::impl {
                     bool uses_custom_row_bounds,
                     host_span<std::vector<size_type> const> row_group_indices);
 
+  /**
+   * @brief Create chunk information and start file reads
+   *
+   * @param row_groups_info vector of information about row groups to read
+   * @param num_rows  Maximum number of rows to read
+   * @return boolean indicating if compressed chunks were found
+   */
+  std::pair<bool, std::vector<std::future<void>>> create_and_read_column_chunks(
+    std::vector<row_group_info> const& row_groups_info, size_type num_rows);
```

Comment:
I'm sure that somebody will request using `host_span` (not me) :smile:

Reply:
Well, I agree so I went ahead and did the work for them.